p2p/host.go

```diff
@@ -405,36 +434,34 @@ func NewHost(cfg HostConfig) (Host, error) {
 	security := security.NewManager(cfg.MaxConnPerIP, int(cfg.MaxPeers), banned)
 	// has to save the private key for host
 	h := &HostV2{
-		h:              p2pHost,
-		pubsub:         pubsub,
-		joined:         map[string]*libp2p_pubsub.Topic{},
-		self:           *self,
-		trustedNodes:   cfg.TrustedNodes,
-		trustedPeerIDs: make(map[libp2p_peer.ID]struct{}),
-		priKey:         key,
-		discovery:      disc,
-		security:       security,
-		onConnections:  ConnectCallbacks{},
-		onDisconnects:  DisconnectCallbacks{},
-		logger:         &subLogger,
-		ctx:            ctx,
-		cancel:         cancel,
-		banned:         banned,
-	}
-
-	for _, addr := range cfg.TrustedNodes {
-		peerAddr, err := ma.NewMultiaddr(addr)
-		if err != nil {
-			subLogger.Error().Err(err).Str("addr", addr).Msg("invalid trusted node addr")
-			continue
-		}
-		info, err := libp2p_peer.AddrInfoFromP2pAddr(peerAddr)
-		if err != nil {
-			subLogger.Error().Err(err).Str("addr", addr).Msg("failed to parse trusted node")
-			continue
-		}
-		h.trustedPeerIDs[info.ID] = struct{}{}
-	}
+		h:               p2pHost,
+		pubsub:          pubsub,
+		joined:          map[string]*libp2p_pubsub.Topic{},
+		self:            *self,
+		trustedNodes:    cfg.TrustedNodes,
+		trustedPeerIDs:  make(map[libp2p_peer.ID]struct{}),
+		trustedMinPeers: cfg.TrustedMinPeers,
+		priKey:          key,
+		discovery:       disc,
+		security:        security,
+		onConnections:   ConnectCallbacks{},
+		onDisconnects:   DisconnectCallbacks{},
+		logger:          &subLogger,
+		ctx:             ctx,
+		cancel:          cancel,
+		banned:          banned,
+	}
+
+	// If no explicit trusted nodes were configured, fall back to boot nodes.
+	if len(h.trustedNodes) == 0 && len(cfg.BootNodes) > 0 {
+		h.trustedNodes = append(h.trustedNodes, cfg.BootNodes...)
```

Comment:
Hm, interesting one, how trusted nodes are connected to the bootnodes?

Reply:
wrong commit, checking now...